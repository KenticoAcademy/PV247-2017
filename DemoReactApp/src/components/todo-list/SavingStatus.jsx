import PropTypes from 'prop-types';
import React from 'react';
import { SavingIcon } from './SavingStatus.styles.jsx';

export class SavingStatus extends React.PureComponent {

    static propTypes = {
        watchedEntity: PropTypes.object.isRequired,
        isSaving: PropTypes.bool.isRequired,
        save: PropTypes.func.isRequired,
    };

    componentWillUpdate(nextProps) {
        if (this.props.watchedEntity !== nextProps.watchedEntity) {
            this.props.save();
        }
    }

    render() {
        const text = this.props.isSaving ? 'Saving' : 'Saved';
        const icon = this.props.isSaving
            ? <SavingIcon alt="saving">_</SavingIcon>
            : <i className="glyphicon glyphicon-check" alt="Saved"/>;

        return (
            <span className="pull-right">
                {icon}{'\u00A0'}{text}
            </span>
        );
    }
}
