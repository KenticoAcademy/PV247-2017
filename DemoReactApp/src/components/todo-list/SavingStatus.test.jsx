import React from 'react';
import renderer from 'react-test-renderer';
import { SavingStatus } from './SavingStatus';

test('shows spinner when saving in progress', () => {
    const component = renderer.create(
        <SavingStatus
            watchedEntity={{}}
            isSaving={true}
            save={() => { /* do nothing */
            }}
        />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});