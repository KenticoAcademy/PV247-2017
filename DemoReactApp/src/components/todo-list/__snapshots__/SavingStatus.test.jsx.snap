// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows spinner when saving in progress 1`] = `
<span
  className="navbar-right sc-bdVaJa liQyis"
>
  <i
    alt="saving"
    className="sc-bwzfXH fdfeVp"
  >
    _
  </i>
  Â 
  Saving
</span>
`;
